shap/plots/dependence.py

```diff
@@ -138,6 +142,15 @@ def convert_name(ind):
     if categorical_interaction and clow != chigh:
         bounds = np.linspace(clow, chigh, chigh - clow + 2)
         color_norm = matplotlib.colors.BoundaryNorm(bounds, colors.red_blue.N)
+        
+    # optionally add noise if feature has less than 20 unique values
+    if (noise > 0) and (len(set(xv)) < 20):
+        if noise > 1: noise = 1
+        xvals = list(set(xv))
+        smallest_diff = np.min([xvals[i] - xvals[i-1] for i in range(1, len(xvals))])
+        noise_amount = noise * smallest_diff
+        for i in range(len(xv)):
```

Comment:
```suggestion
        xv += (np.random.ranf(size = len(xv))*noise_amount) - (noise_amount/2)
```
You could get ride of this for loop and take advantage of vectorized operations here. 
Overall I like your contribution to the dependence plot btw, great idea!

Reply:
Thanks!  Good point on the vectorization.  But I think you should also remove the line just under the FOR statement (cause you replaced it).  Can you do that or let me know how to.  Sorry I'm kinda a git rookie.

Reply:
No problem. See comment below.